saleor/csv/tests/test_products_data.py

```diff
@@ -0,0 +1,1096 @@
+            product_headers.append(f"{attr.slug} (product attribute)")
+        if attr.product_variant_types.exists():
+            variant_headers.append(f"{attr.slug} (variant attribute)")
+
+    expected_headers = product_headers + variant_headers
+    assert attributes_headers == expected_headers
+
+
+def test_get_attributes_headers_lack_of_attributes_ids():
+    # given
+    export_info = {}
+
+    # when
+    attributes_headers = get_attributes_headers(export_info)
+
+    # then
+    assert attributes_headers == []
+
+
+def test_get_warehouses_headers(warehouses):
+    # given
+    warehouse_ids = [warehouses[0].pk]
+    export_info = {"warehouses": warehouse_ids}
+
+    # when
+    warehouse_headers = get_warehouses_headers(export_info)
+
+    # then
+    assert warehouse_headers == [f"{warehouses[0].slug} (warehouse quantity)"]
+
+
+def test_get_warehouses_headers_lack_of_warehouse_ids():
+    # given
+    export_info = {}
+
+    # when
+    warehouse_headers = get_warehouses_headers(export_info)
+
+    # then
+    assert warehouse_headers == []
+
+
+def test_get_export_fields_and_headers_info(
+    warehouses, product_with_multiple_values_attributes
+):
+    # given
+    warehouse_ids = [w.pk for w in warehouses]
+    attribute_ids = [attr.pk for attr in Attribute.objects.all()]
+    export_info = {
+        "fields": [
+            ProductFieldEnum.PRICE.value,
+            ProductFieldEnum.PRICE_OVERRIDE.value,
+            ProductFieldEnum.COLLECTIONS.value,
+            ProductFieldEnum.DESCRIPTION.value,
+        ],
+        "warehouses": warehouse_ids,
+        "attributes": attribute_ids,
+    }
+
+    expected_file_headers = [
+        "id",
+        "price",
+        "product currency",
+        "price override",
+        "variant currency",
+        "collections",
+        "description",
+    ]
+
+    # when
+    export_fields, file_headers, data_headers = get_export_fields_and_headers_info(
+        export_info
+    )
+
+    # then
+    expected_fields = [
+        "id",
+        "price_amount",
+        "currency",
+        "variants__price_override_amount",
+        "variants__currency",
+        "collections__slug",
+        "description",
+    ]
+
+    product_headers = []
+    variant_headers = []
+    for attr in Attribute.objects.all().order_by("slug"):
+        if attr.product_types.exists():
+            product_headers.append(f"{attr.slug} (product attribute)")
+        if attr.product_variant_types.exists():
+            variant_headers.append(f"{attr.slug} (variant attribute)")
+
+    warehouse_headers = [f"{w.slug} (warehouse quantity)" for w in warehouses]
+    excepted_headers = (
+        expected_fields + product_headers + variant_headers + warehouse_headers
+    )
+
+    expected_file_headers += product_headers + variant_headers + warehouse_headers
+
+    assert expected_file_headers == file_headers
+    assert set(export_fields) == set(expected_fields)
+    assert data_headers == excepted_headers
```

Comment:
Also, you could create an `export` directory as we are going to implement `import` also. IMHO - It would be good to separate them now. What do you think?

Reply:
I can create an export directory, but I think it doesn't matter if I do it now or later.

What about file size, I was wondering about splitting it, but this is one file, so there should be a separate file for every function? I am not sure if it's a good idea.

Reply:
- When we will introduce tests for import I assume that we will try to separate them from the export. In that case, you will need to move all files. But it was only an idea so the decision is your ;)

- Based on experience from tests for mutation/queries and splitting the files per:  mutation/query I would say that smaller scope of the test file is better than the huge one.
cc @fowczarek  What do you think?

Reply:
I agree that splitting files to smaller scope increases test readability.

Reply:
In my opinion, working on small files is a lot easier than one big file.

Reply:
I create PR for that:
#5751